### PR TITLE
Add support for gpu op.

### LINKED ISSUE #6: Does it support custom-op with gpu kernel?

### PR DESCRIPTION
**Pull request by yifeif**
- [x]  Add gpu op
- [x]  Update configure.sh for GPU
- [x]  Add basic Makefile support
- [x]  Setup cuda header path, currently the Makefile version requires modifying the cuda header include path
- [x] Setup crosstool for bazel
- [x]  Add BUILD file for GPU
- [x]  Create a docker image for gpu ops
- [x]  Add gpu custom op doc

Fix #6

### CODE REVIEW

File: tensorflow_zero_out/python/ops/zero_out_ops_test.py [hunk 1]
```diff
@@ -20,7 +20,10 @@
 import numpy as np
 
 from tensorflow.python.platform import test
-from zero_out_ops import zero_out
+try:
+  from tensorflow_zero_out.python.ops.zero_out_ops import zero_out
+except ImportError:
+  from zero_out_ops import zero_out
```
Comment (martinwicke) [inline tensorflow_zero_out/python/ops/zero_out_ops_test.py]: Why is this necesary?

Comment (yifeif) [reply tensorflow_zero_out/python/ops/zero_out_ops_test.py]: to make both make and bazel happy.

**Comment by yifeif** [comment]
Thanks for taking a look @martinwicke! If you are talking about ci to make sure the guide always work, we don't have one yet. It can be something useful to set up.

**Comment by martinwicke** [comment]
Mostly I mean a way to test it locally, say you check it out, build it, and
then want to test that it's doing the right thing. You have a test for the
op, but can we add one which ensures that if a GPU is present, it tests the
GPU kernel?

**Comment by yifeif** [comment]
I see. The `time_two` op has both CPU and GPU implementation, and if config=cuda is set (which is set by default in the GPU docker after running ./configure), the GPU version of the ops will be built and ran.

**Comment by martinwicke** [comment]
Ok. Maybe that's enough -- is there a way to tell whether the GPU op is
being run? I.e., if my GPU op isn't registered correctly, will the test
pass?

**Comment by yifeif** [comment]
Yea, there is a python test that loads the op shared library and runs the op.

**Pull request merged**